Add a pure python wrapper fo pybindings.portable_lib

dbort opened the pull request:
Summary:
When installed as a pip wheel, we must import `torch` before trying to import the pybindings shared library extension. This will load libtorch.so and related libs, ensuring that the pybindings lib can resolve those runtime dependencies.

So, add a pure python wrapper that lets us do this when users say `import executorch.extension.pybindings.portable_lib`

We only need this for OSS, so don't bother doing this for other pybindings targets.

Differential Revision: D56317150

mikekgfb reviewed (approved):
Thank you!

larryliu0820 reviewed (approved):
Looks good. I’m thinking is there a way to setup a test? I want to see if we are giving proper error message to user, if they don’t have torch installed properly.

orionr commented on extension/pybindings/portable_lib.py:
```diff
@@ -0,0 +1,34 @@
+# the pybindings shared library extension. This will load libtorch.so and
+# related libs, ensuring that the pybindings lib can resolve those runtime
+# dependencies.
+import torch as _torch
```
Do we rely on ATen for the pybind? I'm guessing including torch here is going to pull all of that in. Okay, but just checking

dbort replied:
We do, so that users can use normal torch tensors for their inputs and outputs, making it easier for them to use.

orionr commented on extension/pybindings/TARGETS:
```diff
@@ -46,8 +46,9 @@ executorch_pybindings(
     cppdeps = PORTABLE_MODULE_DEPS + MODELS_ATEN_OPS_LEAN_MODE_GENERATED_LIB,
-    python_module_name = "portable_lib",
-    types = ["//executorch/extension/pybindings:pybindings_types_gen[portable_lib.pyi]"],
+    # Give this an underscore prefix because it has a pure python wrapper.
```
I must say this is pretty cool

orionr reviewed (approved):
Looks good! Did a real review here.

dbort committed:
Add a pure python wrapper to pybindings.portable_lib (#3137)

Summary:

When installed as a pip wheel, we must import `torch` before trying to import the pybindings shared library extension. This will load libtorch.so and related libs, ensuring that the pybindings lib can resolve those runtime dependencies.

So, add a pure python wrapper that lets us do this when users say `import executorch.extension.pybindings.portable_lib`

We only need this for OSS, so don't bother doing this for other pybindings targets.

Reviewed By: orionr, mikekgfb

Differential Revision: D56317150
